pkg/registry/generic/rest/proxy.go

```diff
@@ -188,7 +209,11 @@ func (h *UpgradeAwareProxyHandler) dialURL() (net.Conn, error) {
 func (h *UpgradeAwareProxyHandler) defaultProxyTransport(url *url.URL) http.RoundTripper {
 	scheme := url.Scheme
 	host := url.Host
-	pathPrepend := strings.TrimRight(url.Path, h.Location.Path)
+	suffix := h.Location.Path
+	if strings.HasSuffix(url.Path, "/") && !strings.HasSuffix(suffix, "/") {
+		suffix += "/"
+	}
```

Comment:
It's ok for the pathPrepend not to end in a "/", since it will only be used in to path.Join with relative URLs in the markup: https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/util/proxy/transport.go#L128-L131. The flipside is that if url.Path is something like /a/b/c/ and suffix (or h.Location.Path) is /b/c, the proper prefix won't be calculated unless I add a '/' to the suffix